### ISSUE TITLE
CA1036 should not report on derived types where base implements IComparable

### ISSUE THREAD

**Issue by twsouthwick** [dotnet/roslyn-analyzers#1671]
#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0]

#### Analyzer

OverrideMethodsOnComparableTypesAnalyzer 

#### Repro steps

```csharp
public class BaseClass<T> : IComparable<T>
     where T : IComparable<T>
{
    public T Value { get; set; }


    public int CompareTo(T other)
    {
        return Value.CompareTo(other);
    }

    public override bool Equals(object obj)
    {
        if (obj is BaseClass<T> other)
        {
            return Value.Equals(other.Value);
        }

        return false;
    }

    public override int GetHashCode() => Value?.GetHashCode() ?? 0;
}

public class DerivedClass<T> : BaseClass<T>
    where T : IComparable<T>
{
}
```

#### Expected behavior

`DerivedClass<T>` should not raise CA1036 as equality is implemented in the base class

#### Actual behavior

`DerivedClass<T>` raises CA1036
